app/models/export.rb

```diff
@@ -0,0 +1,88 @@
+# Adapter to fire off the correct job, track transitions and responses.
+class Export < ApplicationRecord
+  class UnknownExportTypeError < StandardError; end
+
+  include Informed
+
+  belongs_to :snap_application
+  validates :status, inclusion: { in: %i(new queued in_process succeeded failed
+                                         unnecessary) }
+  validates :destination, inclusion: { in: %i(fax email) }
+  before_validation { self.destination = destination.to_sym }
+  before_validation { self.status = status.to_sym }
+
+  default_scope { order(updated_at: :desc) }
+  scope :faxed, -> { where(destination: :fax) }
+  scope :emailed, -> { where(destination: :email) }
+  scope :succeeded, -> { where(status: :succeeded) }
+  scope :completed, -> { where.not(completed_at: nil) }
+  scope :application_ids, -> { pluck(:snap_application_id) }
+  scope :latest, -> { first }
+
+  def self.enqueue_faxes
+    SnapApplication.faxable.find_each do |snap_application|
+      create_and_enqueue(destination: :fax, snap_application: snap_application)
+    end
+  end
+
+  def self.create_and_enqueue(params)
+    create(params).enqueue
+  end
+
+  def self.create_and_enqueue!(params)
+    create!(params).enqueue
+  end
+
+  # Queues up the export job for the given destination
+  def enqueue
+    transaction do
+      save!
```

Comment:
feels a bit weird to me to call `save!` in here since I'd expect it just to enqueue (feels like it is doing two things instead of one) -- thoughts?

Reply:
This protects us from starting a job with a nil export. i.e. `Export.new(...).enqueue` which would break the linkage between the snap application and the particular job to be done.